x-pack/plugin/logsdb/src/yamlRestTest/resources/rest-api-spec/test/80_index_sort_defaults.yml

```diff
@@ -0,0 +1,559 @@
+  - set: { data_streams.0.indices.0.index_name: backing_index }
+
+  - do:
+      indices.get_settings:
+        index: $backing_index
+        include_defaults: true
+  - match: { .$backing_index.settings.index.mode: logsdb }
+  - match: { .$backing_index.settings.index.logsdb.add_host_name_field: null }
+  - match: { .$backing_index.defaults.index.logsdb.add_host_name_field: "false" }
+  - match: { .$backing_index.settings.index.logsdb.sort_on_host_name: null }
+  - match: { .$backing_index.defaults.index.logsdb.sort_on_host_name: "false" }
+  - match: { .$backing_index.defaults.index.sort.field: [ "@timestamp" ] }
+  - match: { .$backing_index.defaults.index.sort.order: [ "desc" ] }
+  - match: { .$backing_index.defaults.index.sort.mode: [ "max" ] }
+  - match: { .$backing_index.defaults.index.sort.missing: [ "_last" ] }
+
+---
+create logsdb data stream with host as text:
+  - requires:
+      test_runner_features: [ "allowed_warnings" ]
+
+  - do:
+      cluster.put_component_template:
+        name: "logsdb-mappings"
+        body:
+          template:
+            settings:
+              mode: "logsdb"
+            mappings:
+              properties:
+                host:
+                  type: "text"
+                "@timestamp":
+                  type: "date"
+                  format: "strict_date_optional_time"
+
+  - do:
+      indices.put_index_template:
+        name: "logsdb-index-template"
+        body:
+          index_patterns: ["logsdb"]
+          data_stream: {}
+          composed_of: ["logsdb-mappings"]
+      allowed_warnings:
+        - "index template [logsdb-index-template] has index patterns [logsdb] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logsdb-index-template] will take precedence during new index creation"
+
+  - do:
+      indices.create_data_stream:
+        name: "logsdb"
+
+  - is_true: acknowledged
+
+  - do:
+      indices.get_data_stream:
+        name: "logsdb"
+        expand_wildcards: hidden
+  - length: { data_streams: 1 }
+  - set: { data_streams.0.indices.0.index_name: backing_index }
+
+  - do:
+      indices.get_settings:
+        index: $backing_index
+        include_defaults: true
+  - match: { .$backing_index.settings.index.mode: logsdb }
+  - match: { .$backing_index.settings.index.logsdb.add_host_name_field: null }
+  - match: { .$backing_index.defaults.index.logsdb.add_host_name_field: "false" }
+  - match: { .$backing_index.settings.index.logsdb.sort_on_host_name: null }
+  - match: { .$backing_index.defaults.index.logsdb.sort_on_host_name: "false" }
+  - match: { .$backing_index.defaults.index.sort.field: [ "@timestamp" ] }
+  - match: { .$backing_index.defaults.index.sort.order: [ "desc" ] }
+  - match: { .$backing_index.defaults.index.sort.mode: [ "max" ] }
+  - match: { .$backing_index.defaults.index.sort.missing: [ "_last" ] }
+---
+create logsdb data stream with host as text and name as double:
```

Comment:
This is a strange setup, but I see how this can work and using host.name as primary sort field.

Reply:
Yeah it's definitely a strange setup, but I found it tested in `30_logsdb_default_mapping`, and I figured it couldn't hurt to test the edge case.